CSUB-366: Deploy on demand self-hosted runners

atodorov opened the pull request:
Description of proposed changes:


Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

atodorov committed:
Provision self-hosted runner on demand and remove it when done

CAGS295 commented on .github/runner.bicep:
```diff
@@ -0,0 +1,871 @@
+
+@description('Type of authentication to use on the Virtual Machine. SSH key is recommended.')
+@allowed([
+  'sshPublicKey'
```
Does this work?
```suggestion
  authenticationType
```

atodorov replied:
It shouldn't. Line 10 is listing the allowed values for this parameter, while line 12 is the current value (+default if not supplied) to be used in deployment. 

Previously the value "password" was also allowed for authentication type. I will hard-code this to sshPublicKey directly instead of providing it as parameter.

atodorov committed:
hard-code VM authentication type to SSH Key

using passwords for SSH is terrible and I don't want to expose this
functionality as a deployment parameter
